src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -115,20 +117,108 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
+                }
+
+                // Verify the expanded template is available.
+                // (Avoid throwing earlier since this is not always needed.)
+                if (expandedTemplateRoot == null)
+                {
+                    throw new ArgumentNullException(nameof(expandedTemplateRoot));
+                }
+
+                string remainingPathAtResourceScope = pathInExpandedTemplate[(resourceWithIndex.Length + 1)..];
+
+                // Get the resource name and type from the expanded template
+                var childResourceName = expandedTemplateRoot.InsensitiveToken($"{resourceWithIndex}.name", InsensitivePathNotFoundBehavior.Null);
+                var childResourceType = expandedTemplateRoot.InsensitiveToken($"{resourceWithIndex}.type", InsensitivePathNotFoundBehavior.Null);
+                if (childResourceName != null && childResourceType != null)
+                {
+                    var childName = childResourceName.Value<string>();
+                    var childType = childResourceType.Value<string>();
+
+                    // Find the resource with this name and type in the expanded template
+                    string resourcePathInOriginalTemplate = GetResourcePath(originalTemplateRoot.InsensitiveToken("resources", InsensitivePathNotFoundBehavior.Null), childName, childType);
+                    
+                    return ResolveLineNumber($"{resourcePathInOriginalTemplate}.{remainingPathAtResourceScope}");
+                }
+            }
             return 1;
         }
+
+        /// <summary>
+        /// Given a list of resources, resource name, and resource type, return the JSON path of that resource.
+        /// </summary>
+        /// <param name="resources">JArray of resources.</param>
+        /// <param name="resourceName">Name of resource.</param>
+        /// <param name="resourceType">Type of resource.</param>
+        /// <returns>The JSON path where the resources is located.</returns>
+        private string GetResourcePath(JToken resources, string resourceName, string resourceType)
+        {
+            if (resources == null)
+            {
+                return null;
+            }
+
+            foreach (var resource in resources.Children())
+            {
+                var thisResourceName = resource.InsensitiveToken("name")?.Value<string>();
+                var thisResourceType = resource.InsensitiveToken("type")?.Value<string>();
+                if (string.Equals(resourceType, thisResourceType, StringComparison.OrdinalIgnoreCase) && string.Equals(resourceName, thisResourceName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return resource.Path;
+                }
+
+                if (resource.InsensitiveToken("resources", InsensitivePathNotFoundBehavior.Null) != null)
+                {
+                    string path = GetResourcePath(resource.InsensitiveToken("resources"), resourceName, resourceType);
```

Comment:
minor: save resource.InsensitiveToken("resources") so it doesn't get fetched twice.